App/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Repositories\Eloquent\TaskRepository;
+use App\Repositories\Eloquent\UserRepository;
+use App\Models\Task;
+use Illuminate\Support\Facades\Auth;
+
+
+class TaskController extends Controller
+{
+  public function __construct()
+  {
+      $this->middleware('auth');
+  }
+
+
+  public function index()
+  {
+      $taskRepository = new TaskRepository();
+      $user = Auth::user();
+      $tasks= $taskRepository->findAllBy('owner_user_id',$user->id);
+
+      return view('tasks.index',compact("tasks"));
+  }
+
+  public function onAssign($id)
+  {
+       $taskRepository = new TaskRepository();
+       $task= $taskRepository->find($id);
+
+       $userRepository = new UserRepository();
+       $user = Auth::user();
+       $users= $userRepository->findAllBy('team_leader_id',$user->id);
+
+       return view('tasks.assign',compact("task","users"));
+   }
+
+   public function assign(Request $request,$id)
+   {
+       $this->validate($request, [
+           'assign_to_user_id' => ['required']
+       ]);
+
+       $taskRepository = new TaskRepository();
+
+       $task = $taskRepository->find($id);
+
+       $task->assign_to_user_id =  $request->get('assign_to_user_id');
+
+       $task = $taskRepository->update($task);
+
+       return redirect('/tasks/index');
+  }
+
+  public function unassign($id)
+  {
+      $taskRepository = new TaskRepository();
+
+      $task = $taskRepository->find($id);
+
+      $task->assign_to_user_id =  null;
+
+      $task = $taskRepository->update($task);
+
+      return redirect('/tasks/index');
+  }
+
+  public function complete($id)
+  {
+      $taskRepository = new TaskRepository();
+
+      $task = $taskRepository->find($id);
+
+      $task->status = 'COMPLETED';
```

Comment:
Always try and use Enums for these statuses. TaskStatusEnum

Reply:
done